mantidproject/mantid
# Fix spurious transmission error with workspace groups

**Report to:** Max at ISIS

**Description of work.**
- This PR fixes some spurious error messages that are issued when you run ReflectometryReductionOneAuto on a workspace group, with transmission corrections enabled. The problem was related to running the same child algorithm multiple times. Now, a new child algorithm is created each time.
- To do this some refactoring was required. Other tidying has also been done to simplify the `processGroups` function and reduce duplicated code
- The documentation regarding processing groups has also been updated because it was out of date.

**To test:**

Run the script in the linked issue and ensure that no red warning messages are displayed in the log.

Fixes #23471 and fixes #29411 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
